pkg/virt-launcher/virtwrap/device/sriov/hostdev.go

```diff
@@ -237,3 +237,33 @@ func buildHostPlugHostDevicesErrorMessage(errors []error) error {
 	}
 	return fmt.Errorf(errorMessageBuilder.String())
 }
+
+func GetHostDevicesToAttach(vmi *v1.VirtualMachineInstance, domainSpec *api.DomainSpec) ([]api.HostDevice, error) {
```

Comment:
Is the dom.attach command is blocking?
If not, what if there are some sriov hostdevices that are in the process of being attached?

Reply:
I just confirmed it with @berrange , https://libvirt.org/html/libvirt-libvirt-domain.html#virDomainAttachDeviceFlags is confirmed to be "blocking".

This whole attachment handling should be protected by the `domainModifyLock` lock, something that was already included (but needs to move closer to here).